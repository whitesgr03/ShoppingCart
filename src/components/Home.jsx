const Home = () => {
	return (
		<div className="home">
			<div className="carousel">
				<button type="button" className="arrow left"></button>
				<div className="content">
					<div className="container">
						<h2 className="title">Casual Summer Collection</h2>
						<button type="button">VIEW MORE</button>
					</div>
				</div>
				<button type="button" className="arrow right"></button>
			</div>
			<div className="info"></div>
			<div className="category"></div>
		</div>
	);
};

export { Home as default };
