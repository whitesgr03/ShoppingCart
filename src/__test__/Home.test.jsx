import { render } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router-dom";

import Home from "../components/Home";

describe("Renders Home Component", () => {
	it("Should return Home DOM", () => {
		const routes = [
			{
				path: "/",
				element: <Home />,
			},
		];

		const router = createMemoryRouter(routes, {
			initialEntries: ["/"],
		});

		const { container } = render(<RouterProvider router={router} />);

		expect(container).toMatchSnapshot();
	});
});
