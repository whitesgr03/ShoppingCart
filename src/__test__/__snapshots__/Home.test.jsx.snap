// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home Component Should return Home DOM 1`] = `
<div>
  <div
    class="home"
  >
    <div
      class="carousel"
    >
      <button
        class="arrow left"
        type="button"
      />
      <div
        class="content"
      >
        <div
          class="container"
        >
          <h2
            class="title"
          >
            Casual Summer Collection
          </h2>
          <a
            class="link"
            href="/shop"
          >
            VIEW MORE
          </a>
        </div>
      </div>
      <button
        class="arrow right"
        type="button"
      />
    </div>
    <div
      class="info"
    />
    <div
      class="category"
    />
  </div>
</div>
`;
